invenio_curations/services/comment.py

```diff
@@ -0,0 +1,258 @@
+
+from .events import CurationCommentEventType
+
+
+class CommentProcessor:
+    """CommentProcessor class.
+
+    Used to create/update curation request comments.
+    """
+
+    _comment_error = {
+        "field": "custom_fields.rdm-curation",
+        "messages": [
+            _("Record saved successfully, but failed to update request comment.")
+        ],
+    }
+
+    def __init__(self, identity, diff_processor):
+        """Constructs."""
+        self._identity = identity
+        self._diff_processor = diff_processor
+
+    def _validate_request(self, request):
+        """Sanity check."""
+        return isinstance(request, dict) and request.get("is_open")
+
+    def _is_comment_hit(self, hit):
+        """Check if a comment was automatically created by this processor."""
+        return hit.get("type") == "C" and hit.get("created_by").get("user") == "system"
+
+    def _get_current_diffs(self, base_data, new_data):
+        """Return the diffs between old and new data.
+
+        Parameters
+        ----------
+        base_data : dict
+            The data to compare against.
+        new_data: dict
+             New data.
+        Returns
+        -------
+        list
+            a list of dictdiffer.diff style tuples
+        """
+        diffs = dictdiffer.diff(base_data, new_data)
+        diff_list = list(diffs)
+
+        if len(diff_list) < 1:
+            return None
+
+        return diff_list
+
+    def _create_new_comment(self, request, content, reference_draft):
+        """Build and send comment create event.
+
+        Parameters
+        ----------
+        request : dict
+            The curation request.
+        content: str
+             The actual comment.
+        reference_draft: str
+             The draft to store for future updates.
+        """
+        payload = {"payload": {"content": content}}
+
+        if reference_draft is not None:
+            payload["payload"]["reference_draft"] = reference_draft
+
+        try:
+            current_events_service.create(
+                self._identity, request["id"], payload, CurationCommentEventType()
+            )
+        except Exception as e:
+            print_exception(e)
+            return self._comment_error
+
+    def _update_existing_comment(self, new_data, crt_comment_event):
+        """Build and send comment update event.
+
+        Parameters
+        ----------
+        new_data : str
+            The updated comment.
+        crt_comment_event: dict
+            The existing comment.
+        """
+        if crt_comment_event is None:
+            return self._comment_error
+
+        payload = {"payload": {"content": new_data}}
+
+        try:
+            current_events_service.update(
+                self._identity,
+                crt_comment_event.get("id"),
+                payload,
+                revision_id=crt_comment_event.get("revision_id"),
+            )
+        except Exception as e:
+            print_exception(e)
```

Comment:
i would avoid calling `print()`, and rather go through the app's configured logger:
```suggestion
            current_app.logger.warn(e)
```